### PR TITLE
Authorization handle

### LINKED ISSUE #19: Integration with el padro municipal
## Description

A request to el Padro Municipal API is required to authorize the user and validate it as citizen of the location.

The endpoint has the following format:

`PADRO_ENDPOINT_URL?nif=1111111X&data=d/m/YYYY`

**Success answer:**

```
{
  exists: true,
  id:12345667888887655
}
```

**Fail answer:**

```
{
  exists: false,
  id:
}
```

The citizen unique identifier needs to be stored in the `metadata` field of the Authorization object.

## Dev details

These data is managed in Decidim through **Authorization Handlers**: https://github.com/MarsBased/decidim-vilanova/blob/master/app/services/example_authorization_handler.rb

We need to remove the example authorization handler and create a new one to integrate with el padró municipal de Vilanova.

In the Decidim initializer different Authoritzation Handlers are enabled.

API endpoint URL needs to be managed via ENV variables. secrets.yml file must be used the way it's being used now.

The project is using MiniTest to test the application instead of Rspec. We'll surely need to adapt the `it_behaves_like "an authorization handler"` helper.

### PR DESCRIPTION
**Pull request by artero**
closes #19 

- Add Vilanova authorization handle
- Fix es locale
- Add yarn config

### CODE REVIEW

**Comment by artero** [comment]
I don't know why this branch is not referred in Waffle.

What I did wrong?

**Comment by xredo** [comment]
Closes #34

**Comment by xredo** [comment]
Closes #19

File: config/initializers/assets.rb [hunk 1]
```diff
@@ -9,3 +9,4 @@
 # Precompile additional assets.
 # application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
 # Rails.application.config.assets.precompile += %w( search.js )
+Rails.application.config.assets.paths << Rails.root.join("vendor", "assets", "node_modules")
```
Comment (xredo) [inline config/initializers/assets.rb]: Single quotes

File: app/services/vilanova_authorization_handler.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  attribute :document_number, String
+  attribute :birthday, String
+
+  validates :document_number, :birthday, presence: true
+  validate :censed_and_add_metadata
+
+  def censed_and_add_metadata
+    errors.add(:document_number, I18n.t('errors.messages.not_censed')) unless cense
+    false
```
Comment (xredo) [inline app/services/vilanova_authorization_handler.rb]: The return value is not important in custom validators. You can remove the false.

http://guides.rubyonrails.org/active_record_validations.html#performing-custom-validations

File: app/services/cense_api.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require 'httparty'
+
+class CenseApi
+
+  def self.search_user_vilanova_id(handle)
+    response_parsed = JSON.parse(
+      HTTParty.get('http://aj_ws.ajuntament.org/comprovaempadronament/default.aspx',
+                   query: { 'nif' => handle.document_number,
```
Comment (xredo) [inline app/services/cense_api.rb]: Does it work using symbol hash syntax? If it works, we should use this.

File: app/services/cense_api.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+                            'data' => handle.birthday })
+    )
+
+    return false if response_parsed['id'].blank? || response_parsed['exists'] == false
```
Comment (xredo) [inline app/services/cense_api.rb]: The search_user_vilanova_id is returning 2 different types. When I call the search_user_vilanova_id i'm expecting to receive an ID or nil if I there is no user vilanova id.

This way you don't need to use in another place of your code:

```
def unique_id
   cense || nil
end
```

File: .gitignore [hunk 5]
```diff
@@ -18,9 +18,13 @@
...
+
 # Ignore application configuration
 /config/application.yml
 
 # Ignore env variables
 .env
 .env.*
+
+# Ignore yarn libraries
+vendor/assets/node_modules
```
Comment (xredo) [inline .gitignore]: If you ignore yarn libraries, then we need to install them in the production environment. We need to tune the Docker file. I wouldn't ignore them for simplicity sake. It's not a problem for now.

**Comment by artero** [comment]
I did all changes. 

Sorry I didn't do your suggested implementation because I had problems with the tests and I thought is better finish early.

**Pull request merged**